packages/beacon-node/src/execution/engine/http.ts

```diff
@@ -493,75 +498,85 @@ export class ExecutionEngineHttp implements IExecutionEngine {
+      this.logger.error(error);
+      throw Error(error);
+    }
+
+    return (response as EngineApiRpcReturnTypes["engine_getBlobsV1"]).map(deserializeBlobAndProofs);
+  }
+
+  private async getBlobsV2(versionedHashesHex: string[]) {
+    const response = await this.rpc
+      .fetchWithRetries<EngineApiRpcReturnTypes["engine_getBlobsV2"], EngineApiRpcParamTypes["engine_getBlobsV2"]>({
+        method: "engine_getBlobsV2",
+        params: [versionedHashesHex],
+      })
+      .catch((e) => {
+        if (e instanceof ErrorJsonRpcResponse && parseJsonRpcErrorCode(e.response.error.code) === "Method not found") {
+          this.logger.debug("disabling engine_getBlobsV2 api call since engine responded with method not available", {
+            retryTimeout: GET_BLOBS_RETRY_TIMEOUT,
           });
         }
         throw e;
       });
 
     // engine_getBlobsV2 does not return partial responses. It returns an empty array if any blob is not found
     // TODO: Spec says to return null if any blob is not found, but reth and nethermind return empty arrays as of peerdas-devnet-6
-    const invalidLength =
-      method === "engine_getBlobsV2"
-        ? response && response.length !== 0 && response.length !== versionedHashes.length
-        : !response || response.length !== versionedHashes.length;
+    const invalidLength = response && response.length !== 0 && response.length !== versionedHashesHex.length;
 
     if (invalidLength) {
-      const error = `Invalid ${method} response length=${response?.length ?? "null"} versionedHashes=${versionedHashes.length}`;
+      const error = `Invalid engine_getBlobsV2 response length=${response?.length ?? "null"} versionedHashes=${versionedHashesHex.length}`;
       this.logger.error(error);
       throw Error(error);
     }
 
-    // engine_getBlobsV2 returns a list of cell proofs per blob, whereas engine_getBlobsV1 returns one proof per blob
-    switch (method) {
-      case "engine_getBlobsV1":
-        return (response as EngineApiRpcReturnTypes[typeof method]).map(deserializeBlobAndProofs);
-      case "engine_getBlobsV2": {
-        const castResponse = response as EngineApiRpcReturnTypes[typeof method];
-        // TODO(fulu): Spec says to return null if any blob is not found, but reth and nethermind return empty arrays as of peerdas-devnet-6
-        // TODO(fulu): Erigon returns array with `null` values
-        // biome-ignore lint/suspicious/noExplicitAny: <explanation>
-        if (castResponse === null || castResponse.length === 0 || castResponse?.includes(null as any)) return null;
-        return castResponse.map(deserializeBlobAndProofsV2);
-      }
-    }
+    const castResponse = response as EngineApiRpcReturnTypes["engine_getBlobsV2"];
+    // TODO(fulu): Spec says to return null if any blob is not found, but reth and nethermind return empty arrays as of peerdas-devnet-6
+    // TODO(fulu): Erigon returns array with `null` values
```

Comment:
```suggestion
    // SPEC: EL clients MUST return null in case of any missing or older version blobs.
    // https://github.com/ethereum/execution-apis/blob/main/src/engine/osaka.md#specification-1
```
EL clients return `null` when the blobs are missing.
[Erigon PR](https://github.com/erigontech/erigon/pull/15420/files#diff-66b63be03f24b8f8191891c8f447facaa7c3c8e1332ca82f2bed23e3c2279553R958)
[Nethermind PR](https://github.com/NethermindEth/nethermind/pull/8417/commits/d0bdc3f245b211c0316e98d407205de76322d84a#diff-80cb49f35f9081c1426de2d16d5500cbce131fdc2535408cacc9aaa8ab93e285R18)
[Reth PR](https://github.com/paradigmxyz/reth/pull/16271/files#diff-f079abe52ebfa4aa355adbd9122d039f266286b297001040e55a285ea5aecdf9R1081)
Should be approved by the EL teams.

Reply:
All the three teams approved returning `null`.